### PR TITLE
Move E2E tests workflow from CircleCI to GH Actions

### PR DESCRIPTION
**Pull request by michalinacienciala**
With this commit/PR we are switching from using CircleCI for performing
of E2E tests to Github Actions. The old CircleCI workflow used for E2E
testing had been broken for some time, so apart from just moving the
steps "one to one" from CCI to GHA, failing steps needed to be fixed.
Some of the changes include:
- creation of config file for GH Actions workflow performing E2E tests
- removal of config file for CircleCI job performing E2E tests
- switching to higher version of Node.js for E2E tests (from `14.3.0`
  to `14.17.0`)
- adjusting steps of the install scripts to changes introduced in
  https://github.com/keep-network/local-setup/pull/76

### CODE REVIEW

**Comment by michalinacienciala** [comment]
Note: In order for the workflow to run on cron schedule, it needs to be checked in to master. But if you want to see if workflow passes, you can take a look here: https://github.com/keep-network/local-setup/actions/runs/893528342/workflow - this is a run which was executed before removal of 
```
  push: # TODO: remove
    branches:
      - 'rfc-18/**'
```

**Comment by michalinacienciala** [comment]
Hmm, now when I think about it, it may be worth to add `workflow_dispatch` triggering condition to the workflow, in case we would like to execute workflow at different time than specified by cron... I will add this.

File: .github/workflows/e2e-local.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          ./install-go.sh
+          echo "Using Go: $(go version)"
+
+      - name: Set GOPATH variable
+        run: echo "GOPATH=$HOME/go" >> $GITHUB_ENV
+
+      - name: Append PATH variable
+        run: |
+          echo "/usr/local/go/bin" >> $GITHUB_PATH
+          echo "$GOPATH/bin" >> $GITHUB_PATH
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:51]: Can we merge these three?
```suggestion
      - name: Install Go
        working-directory: ./deployments/local-setup-environment/provisioning/
        run: |
          sudo rm -rf /usr/local/go
          ./install-go.sh
          echo "GOPATH=$HOME/go" >> $GITHUB_ENV
          echo "/usr/local/go/bin" >> $GITHUB_PATH
          echo "$GOPATH/bin" >> $GITHUB_PATH
          echo "Using Go: $(go version)"
```

File: .github/workflows/e2e-local.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        working-directory: ./deployments/local-setup-environment/provisioning/
+        run: ./install-solidity.sh
+
+      - name: Install Protobuf
+        working-directory: ./deployments/local-setup-environment/provisioning/
+        run: ./install-protobuf.sh
+
+      - name: Initialize and run Geth
+        run: |
+          ls -l ethereum/data
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:67]: Can we remove this `ls`?

File: .github/workflows/e2e-local.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          ./initialize-geth.sh
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:68]: The step name says Initialize and **_run_** Geth. Can we move run geth command execution from `Run install script` to here?

File: .github/workflows/e2e-local.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+      - name: Run Bitcoin Core and ElectrumX
+        run: ./run-bitcoin.sh &
+
+      - name: Run install script
+        run: |
+          ls -l ethereum/data
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:75]:
```suggestion
```

File: .github/workflows/e2e-local.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          ./run-geth.sh &
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:76]: Let's move it up to `Initialize and run Geth` step.

File: install-e2e-test.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -e
+
+echo "Running npm install..."
+cd e2e
+npm install
+cd ../keep-ecdsa/solidity
+npm install --save-dev sha3
```
Comment (nkuba) [inline install-e2e-test.sh:9]: If this dependency is required it should be included in keep-ecdsa's `package.json`, we should not need to install it manually.

File: .github/workflows/e2e-local.yml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+      - name: Initialize and run Geth
+        run: |
+          ls -l ethereum/data
+          ./initialize-geth.sh
+
+      - name: Run Bitcoin Core and ElectrumX
+        run: ./run-bitcoin.sh &
+
+      - name: Run install script
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:73]: Don't we want to executed `pre-install.sh` script before running `install.sh`?

**Review by nkuba — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-local.yml (hunk 1 above)
Comment (michalinacienciala) [reply .github/workflows/e2e-local.yml]: Only partially. 
Adding GOPATH to the list of environment variables available to GITHUB can be performed in the same step in which Go gets installed. However, we cannot include in that step the command which appends the `$GITHUB_PATH` with `$GOPATH/bin` value, as the `$GOPATH` variable would not be available at this stage yet.
I merged the `Set GOPATH variable` step with `Install Go`step, see here: https://github.com/keep-network/local-setup/commit/047700f8a1643a0605fd19670a4e9216a465d286

File: .github/workflows/e2e-local.yml (hunk 2 above)
Comment (michalinacienciala) [reply .github/workflows/e2e-local.yml]: Sure, see 19f9bc71b0f3ebcaeba1d1eedbbcb53247c08246.

File: .github/workflows/e2e-local.yml (hunk 3 above)
Comment (michalinacienciala) [reply .github/workflows/e2e-local.yml:70]: Sure, see 19f9bc71b0f3ebcaeba1d1eedbbcb53247c08246.

File: .github/workflows/e2e-local.yml (hunk 4 above)
Comment (michalinacienciala) [reply .github/workflows/e2e-local.yml]: Done, see 19f9bc71b0f3ebcaeba1d1eedbbcb53247c08246.

File: .github/workflows/e2e-local.yml (hunk 5 above)
Comment (michalinacienciala) [reply .github/workflows/e2e-local.yml]: Done, see 19f9bc71b0f3ebcaeba1d1eedbbcb53247c08246.

File: .github/workflows/e2e-local.yml (hunk 7 above)
Comment (michalinacienciala) [reply .github/workflows/e2e-local.yml:79]: You're right, we do. See 6036ee60fdadc105f0e26e9a1a92ba7f81a8ee04.

File: install-e2e-test.sh (hunk 6 above)
Comment (michalinacienciala) [reply install-e2e-test.sh:17]: The manual install of `sha3` is needed for successful running of `npm link` in `local-setup/keep-ecdsa/solidity`. I tried replacing `npm install --save-dev sha3` with `npm rebuild`, but it did not help for the problem. I brought back the original solution.

**Comment by michalinacienciala** [comment]
Successful run after applying requested changes: https://github.com/keep-network/local-setup/actions/runs/896952892 (after that the only another commit was the removal of triggering on push).

File: .github/workflows/e2e-local.yml (hunk 1 above)
Comment (nkuba) [reply .github/workflows/e2e-local.yml]: Maybe we could do it in one step without using `GOPATH`?
```
echo "GOPATH=$HOME/go" >> $GITHUB_ENV
echo "$HOME/go/bin" >> $GITHUB_PATH
echo "/usr/local/go/bin" >> $GITHUB_PATH
```

**Review by nkuba — COMMENTED** [review]
Minor comments.

**Comment by michalinacienciala** [comment]
> Maybe we could do it in one step without using GOPATH?

Right, done and tested on wip branch (https://github.com/keep-network/local-setup/runs/2726571500?check_suite_focus=true), works ok (with incorrectly set variables, Step with Protobuf would fail). Commited in d61b8ce2d6ee450ac1cc25a16b377328601329f0.

File: .github/workflows/e2e-local.yml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+          
+      - name: Install Go
+        working-directory: ./deployments/local-setup-environment/provisioning/
+        run: |
+          sudo rm -rf /usr/local/go
+          ./install-go.sh
+          echo "Using Go: $(go version)"
+          echo "GOPATH=$HOME/go" >> $GITHUB_ENV
+          echo "$HOME/go/bin" >> $GITHUB_PATH
+          echo "/usr/local/go/bin" >> $GITHUB_PATH
```
Comment (nkuba) [inline .github/workflows/e2e-local.yml:46]: I'm not sure we need this both... Isn't it enough to add just `$HOME/go/bin` to GITHUB_PATH?

**Review by nkuba — APPROVED** [review]

**Pull request merged**